File: Arrays/Question 14/two_d_translate.rb
def two_d_translate(arr)
    new_arr = []
  
    arr.each do |subArray|
      ele = subArray[0]
      num = subArray[1]
  
      num.times { new_arr << ele }
    end
  
    return new_arr
  end
  
  arr_1 = [
    ['boot', 3],
    ['camp', 2],
    ['program', 0]
  ]
  
  print two_d_translate(arr_1) # => [ 'boot', 'boot', 'boot', 'camp', 'camp' ]
  puts
  
  arr_2 = [
    ['red', 1],
    ['blue', 4]
  ]
  
  print two_d_translate(arr_2) # => [ 'red', 'blue', 'blue', 'blue', 'blue' ]
  putsa

File: Question 14/two_d_translate.js
function two_d_translate(arr){
    let newArr= '';
    let array =[];
  
    for(let i =0; i< arr.length; i++){
      for(let j=0; j< arr[i].length; j = j+2){
        let ele = arr[i][j];
        let num = arr[i][j+1];
  
        if (num === 0){
          continue;
        }
        array[i] = Array(num).fill(ele);
      }
      
    }
    newArr = array.toString().split(',');
    return newArr;
  }
  
  arr_1 = [
    ['boot', 3],
    ['camp', 2],
    ['program', 0]
  ]
  
  console.log(two_d_translate(arr_1));